### PR TITLE
[CoreBundle] Data list auto delete

### PR DESCRIPTION
**Pull request by ngodfraind**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | yes/no
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any

### CODE REVIEW

**Comment by ngodfraind** [comment]
@ping elorfin

File: main/core/Resources/modules/layout/list/containers/data-list.jsx [hunk 1]
```diff
@@ -226,6 +246,22 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
...
+    props.actions.push({
+      icon: 'fa fa-fw fa-trash-o',
+      label: t('delete'),
+      action: (rows) => dispatchProps.deleteItems(
+        rows,
+        stateProps.modalDeleteTitle(rows),
+        stateProps.modalDeleteQuestion(rows)
+      ),
+      dangerous: true,
+      autoDelete: true
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/list/containers/data-list.jsx]: What is this prop ?

File: main/core/Resources/modules/layout/list/containers/data-list.jsx [hunk 2]
```diff
@@ -84,9 +89,15 @@ function mapStateToProps(state, ownProps) {
...
     totalResults: listSelect.totalResults(listState),
     async: listSelect.isAsync(listState),
+    deletable: listSelect.isDeletable(listState),
     queryString: listSelect.queryString(listState)
   }
 
+  if (newProps.deletable) {
+    newProps.modalDeleteTitle = listSelect.deleteBulk(listState).title
+    newProps.modalDeleteQuestion = listSelect.deleteBulk(listState).question
+  }
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/list/containers/data-list.jsx]: You will never need it inside the component. So you can remove it and just use the selectors in `mergeProps`.

File: main/core/Resources/modules/layout/list/selectors.js [hunk 3]
```diff
@@ -8,14 +8,22 @@ const isSelectable = (listState) => typeof listState.selected !== 'undefined'
...
+const data          = (listState) => listState.data
+const totalResults  = (listState) => listState.totalResults
+const filters       = (listState) => listState.filters || []
+const sortBy        = (listState) => listState.sortBy || {}
+const selected      = (listState) => listState.selected || []
+const pageSize      = (listState) => listState.pageSize || -1
+const currentPage   = (listState) => listState.page || 0
+
+
+const deleteBulkUrl = createSelector(
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/list/selectors.js]: I would rename it in `deleteUrl`.

File: main/core/Resources/modules/layout/list/containers/data-list.jsx [hunk 4]
```diff
@@ -226,6 +246,22 @@ function mergeProps(stateProps, dispatchProps, ownProps) {
     }
   }
 
+  //this might not be where it should be defined but it works
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/list/containers/data-list.jsx]: I'm not sure either but let it here for now.

File: main/core/Resources/modules/layout/list/selectors.js [hunk 5]
```diff
@@ -8,14 +8,22 @@ const isSelectable = (listState) => typeof listState.selected !== 'undefined'
...
-const data         = (listState) => listState.data
-const totalResults = (listState) => listState.totalResults
-const filters      = (listState) => listState.filters || []
-const sortBy       = (listState) => listState.sortBy || {}
-const selected     = (listState) => listState.selected || []
-const pageSize     = (listState) => listState.pageSize || -1
-const currentPage  = (listState) => listState.page || 0
+const fetchUrl      = (listState) => listState.fetchUrl
+const deleteBulk    = (listState) => listState.deleteBulk || {}
+const isDeletable   = (listState) => typeof listState.deleteBulk !== Boolean(listState.deleteBulk)
```
Comment (Elorfin) [inline main/core/Resources/modules/layout/list/selectors.js]: Weird check.

File: main/core/Resources/translations/platform.fr.json [hunk 6]
```diff
@@ -1482,5 +1482,7 @@
     "minimize": "Réduire",
     "display_parameters": "Paramètres d'affichage",
     "add_comment": "Commenter",
-    "keywords": "Mots clés"
+    "keywords": "Mots clés",
+    "objects_delete_title": "Supprimer des objets",
+    "object_delete_question": "{1} Êtes-vous sur de vouloir supprimer l'objet sélectionné ?|[2,Inf[ Êtes-vous sur de vouloir supprimer les %count% objets sélectionnés ?"
```
Comment (Elorfin) [inline main/core/Resources/translations/platform.fr.json]: Êtes-vous sûr

**Pull request merged**